docker/base.dockerfile

```diff
@@ -1,43 +1,30 @@
-FROM ubuntu:jammy
-
-ENV NODE_VERSION=16.20.0
-
-# install required packages
-RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install -y \
-    curl \
-    software-properties-common \
-    gettext \
-    git \
-    git-lfs \
-    psmisc \
-    python3 \
-    python3-pip \
-    python3-sphinx
-
-# add yarn ppa
-RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-
-# install nodejs and yarn
-RUN apt-get update && \
-    ARCH=$(dpkg --print-architecture) && \
-    curl -sSO https://deb.nodesource.com/node_16.x/pool/main/n/nodejs/nodejs_$NODE_VERSION-1nodesource1_$ARCH.deb && \
-    dpkg -i ./nodejs_$NODE_VERSION-1nodesource1_$ARCH.deb && \
-    rm nodejs_$NODE_VERSION-1nodesource1_$ARCH.deb && \
-    apt-get install yarn
-
-RUN git lfs install
-
-# Check if symbolic links exist before creating them
-RUN if [ ! -f /usr/bin/python ]; then ln -s /usr/bin/python3 /usr/bin/python; fi \
- && if [ ! -f /usr/bin/pip ]; then ln -s /usr/bin/pip3 /usr/bin/pip; fi
-
-# copy Kolibri source code into image
-COPY . /kolibri
-
-# do the time-consuming base install commands
-RUN cd /kolibri \
-    && pip3 install -r requirements/dev.txt \
-    && pip3 install -r requirements/test.txt \
-    && yarn install --network-timeout 100000
+# This Dockerfile serves as a base image for Kolibri installations. It installs python, required OS packages, node, npm, yarn and python dependencies.
+
+# Declare python version argument.
+ARG PYTHON_VARIANT="3.9-slim"
+FROM python:${PYTHON_VARIANT}
+
+# Declare node major version argument.
+ARG NODE_MAJOR_VERSION="16"
+
+# Install only the absolutely-required packages to run Kolibri -- Git, Git-LFS, ip, ps and Node (plus npm).
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y ca-certificates curl gnupg git git-lfs procps iproute2 \
+  && mkdir -p /etc/apt/keyrings \
+  && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
```

Comment:
There isn't much benefit to setting up nodesource as a apt-repo inside the image. Also, there are better ways to handle this in a docker file, to better take advantage of caching.

Reply:
Nodesource suggested setting up apt-repo so that `apt update` and `apt install` can figure out nodesource repo.
https://github.com/nodesource/distributions?tab=readme-ov-file#installation-instructions

Regarding caching, I tried to keep the commands in one RUN command to make best use of caching. If there's more we can do to further improve on that then please let me know and I will do it.

Reply:
Right, they'd suggest an apt-repo, but this is a container! It will likely only ever be installed once so it'll never use it for checking for updates!

For example, if the node.js version changes, this installs `ca-certificates curl gnupg git git-lfs procps iproute2` all over again. So no this isn't making the best use of caching.

Reply:
> Right, they'd suggest an apt-repo, but this is a container! It will likely only ever be installed once so it'll never use it for checking for updates!

So without apt-repo, how should we install node?

Reply:
Your changes here modified how we were doing it before, which was not to use the apt-repo.

Reply:
Yeah right, I will fix that with our refactor of base & dev dockerfiles.